src/lexicon/_private/providers/arvancloud.py

```diff
@@ -0,0 +1,267 @@
+            data=json.dumps(data),
+            headers=headers,
+        )
+
+        # if the request fails for any reason, throw an error.
+        response.raise_for_status()
+        return response.json()
+
+    # Takes record input and puts it into a format the Arvancloud API supports
+    def _to_r1c_record(self, rtype, name, content):
+        if rtype in ["LOC", "SSHFP"]:
+            raise NotImplementedError(f"The {rtype} isn't available for Arvancloud")
+        output = {"type": rtype}
+
+        if name is not None:
+            output["name"] = self._relative_name(name)
+
+        content_split = content.split(" ")
+
+        if rtype in ["A", "AAAA"]:
+            if len(content_split) == 1:
+                output.update(
+                    {
+                        "value": [{'ip': content_split[0]}],
+                    }
+                )
+            else:
+                output.update(
+                    {
+                        "value": [{'ip': ip, 'port': int(port), 'weight': int(weight)} for ip, port, weight in content_split],
+                    }
+                )
+
+        output.update(
+            {
+                "CNAME": (lambda: {
+                    "value": {
+                        "host": content_split[0],
+                        "host_header": content_split[1] if len(content_split) == 2 else "source",
+                    }}),
+                "ANAME": (lambda: {
+                    "value": {
+                        "location": content_split[0],
+                        "host_header": content_split[1],
+                    }}),
+                "MX": (lambda: {
+                    "value": {
+                        "host": content_split[0],
+                        "priority": content_split[1],
+                    }
+                }),
+                "NS": (lambda: {"value": {"host": content}}),
+                "PTR": (lambda: {"value": {"domain": content}}),
+                "SRV": (
+                    lambda: {
+                        "value": {
+                            "target": content_split[0],
+                            "port": content_split[1],
+                            "weight": content_split[2],
+                            "priority": content_split[3],
+                        }
+                    }
+                ),
+                "TXT": (lambda: {"value": {"text": content}}),
+                "SPF": (lambda: {"value": {"text": content}}),
+                "DKIM": (lambda: {"value": {"text": content}}),
+            }.get(rtype, lambda: {})()
+        )
+
+        return output
+
+    # Takes record's value and puts it into a format the lexicon supports
+    def _parse_r1c_response(self, rtype: str, input: str):
+        output = {}
+
+        output.update(
+            {
+                "A": (lambda: {"value": ", ".join([" ".join(str(value) if value is not None and value != "" else "" for value in item.values()) for item in input])}),
```

Comment:
See my discussion above about how records should be represented, but at the end I think it should simply be `"A": (lambda: {"value": input['ip']})`.

Also current implementation triggers mypy: the `input` field of this method is certainly not a string.

Finally, what about the other supported records, like `AAAA`?